File: packages/native_plugin/android/src/main/kotlin/com/example/native_plugin/NativePlugin.kt
package com.example.native_plugin

import android.content.Context
import android.os.Handler
import android.os.Looper
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result
import kotlin.random.Random

/** NativePlugin */
class NativePlugin: FlutterPlugin, MethodCallHandler {
  /// The MethodChannel that will the communication between Flutter and native Android
  private lateinit var methodChannel : MethodChannel
  
  /// Event Channels for streaming data
  private lateinit var randomNumberEventChannel: EventChannel
  
  /// Context for accessing system services
  private lateinit var context: Context
  
  /// Stream handlers for each Event Channel
  private val randomNumberStreamHandler = RandomNumberStreamHandler()

  override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
    context = flutterPluginBinding.applicationContext
    
    // Method Channel
    methodChannel = MethodChannel(flutterPluginBinding.binaryMessenger, "native_plugin")
    methodChannel.setMethodCallHandler(this)
    
    // Event Channels with individual stream handlers
    randomNumberEventChannel = EventChannel(flutterPluginBinding.binaryMessenger, "native_plugin/random_numbers")
    randomNumberEventChannel.setStreamHandler(randomNumberStreamHandler)
  }

  override fun onMethodCall(call: MethodCall, result: Result) {
    when (call.method) {
      "getPlatformVersion" -> {
        result.success("Android ${android.os.Build.VERSION.RELEASE}")
      }
      "calculateSum" -> {
        val a = call.argument<Int>("a") ?: 0
        val b = call.argument<Int>("b") ?: 0
        result.success(a + b)
      }
      "getDeviceInfo" -> {
        val deviceInfo = mapOf(
          "brand" to android.os.Build.BRAND,
          "model" to android.os.Build.MODEL,
          "version" to android.os.Build.VERSION.RELEASE,
          "sdk" to android.os.Build.VERSION.SDK_INT,
          "manufacturer" to android.os.Build.MANUFACTURER
        )
        result.success(deviceInfo)
      }
      else -> {
        result.notImplemented()
      }
    }
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    methodChannel.setMethodCallHandler(null)
    randomNumberEventChannel.setStreamHandler(null)
    
    // Stop all monitoring
    stopAllMonitoring()
  }

  private fun stopAllMonitoring() {
    randomNumberStreamHandler.stopStream()
  }

  // Random Number Stream Handler
  inner class RandomNumberStreamHandler : EventChannel.StreamHandler {
    private var eventSink: EventChannel.EventSink? = null
    private var runnable: Runnable? = null
    private val handler = Handler(Looper.getMainLooper())

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
      eventSink = events
      startStream()
    }

    override fun onCancel(arguments: Any?) {
      eventSink = null
      stopStream()
    }

    private fun startStream() {
      runnable = object : Runnable {
        override fun run() {
          if (eventSink != null) {
            val randomNumber = Random.nextInt(1, 101) // 1-100
            eventSink?.success(randomNumber)
            handler.postDelayed(this, 1000) // Send every second
          }
        }
      }
      handler.post(runnable!!)
    }

    fun stopStream() {
      runnable?.let { handler.removeCallbacks(it) }
      runnable = null
    }
  }

}

File: packages/native_plugin/ios/Classes/NativePlugin.swift
import Flutter
import UIKit
import Foundation

public class NativePlugin: NSObject, FlutterPlugin, FlutterStreamHandler {
  /// Method Channel
  private var methodChannel: FlutterMethodChannel?
  
  /// Event Channels
  private var randomNumberEventChannel: FlutterEventChannel?
  
  /// Event sinks for streaming data
  private var randomNumberEventSink: FlutterEventSink?
  
  /// Timers for periodic updates
  private var randomNumberTimer: Timer?

  public static func register(with registrar: FlutterPluginRegistrar) {
    let instance = NativePlugin()
    
    // Method Channel
    let methodChannel = FlutterMethodChannel(name: "native_plugin", binaryMessenger: registrar.messenger())
    registrar.addMethodCallDelegate(instance, channel: methodChannel)
    instance.methodChannel = methodChannel
    
    // Event Channels
    instance.randomNumberEventChannel = FlutterEventChannel(name: "native_plugin/random_numbers", binaryMessenger: registrar.messenger())
    instance.randomNumberEventChannel?.setStreamHandler(instance)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("iOS " + UIDevice.current.systemVersion)
      
    case "calculateSum":
      let args = call.arguments as? [String: Any]
      let a = args?["a"] as? Int ?? 0
      let b = args?["b"] as? Int ?? 0
      result(a + b)
      
    case "getDeviceInfo":
      let deviceInfo: [String: Any] = [
        "brand": "Apple",
        "model": UIDevice.current.model,
        "version": UIDevice.current.systemVersion,
        "sdk": UIDevice.current.systemVersion,
        "manufacturer": "Apple",
        "name": UIDevice.current.name,
        "identifier": UIDevice.current.identifierForVendor?.uuidString ?? "unknown"
      ]
      result(deviceInfo)
      
    default:
      result(FlutterMethodNotImplemented)
    }
  }
  
  // MARK: - FlutterStreamHandler
  
  public func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
    if let channelName = arguments as? String {
      switch channelName {
      case "random_numbers":
        randomNumberEventSink = events
        startRandomNumberStream()
      default:
        break
      }
    }
    return nil
  }
  
  public func onCancel(withArguments arguments: Any?) -> FlutterError? {
    if let channelName = arguments as? String {
      switch channelName {
      case "random_numbers":
        randomNumberEventSink = nil
        stopRandomNumberStream()
      default:
        break
      }
    }
    return nil
  }
  
  // MARK: - Random Number Stream
  
  private func startRandomNumberStream() {
    randomNumberTimer = Timer.scheduledTimer(withTimeInterval: 1.0, repeats: true) { [weak self] _ in
      guard let self = self else { return }
      if let eventSink = self.randomNumberEventSink {
        let randomNumber = Int.random(in: 1...100)
        eventSink(randomNumber)
      }
    }
  }
  
  private func stopRandomNumberStream() {
    randomNumberTimer?.invalidate()
    randomNumberTimer = nil
  }
}
